data/plugins/autoreload.lua

```diff
@@ -30,13 +30,31 @@ local function update_time(doc)
 end
 
 local function reload_doc(doc)
-  doc:reload()
+  local content, crlf = doc:load_content(doc.abs_filename)
+  doc.crlf = crlf
+
+  local selections = {}
+  for idx, line1, col1, line2, col2 in doc:get_selections() do
+    selections[idx] = { line1, col1, line2, col2 }
+  end
+  
+  local len = #doc.lines
+  doc:remove(1, 1, len, #doc.lines[len])
```

Comment:
You should be able to do:
```suggestion
  doc:remove(1, 1, math.huge, math.huge)
```